crawl-ref/source/state.h

```diff
@@ -139,6 +149,12 @@ struct game_state
     // Should flushing a nonempty key buffer error or crash? Used for tests.
     bool nonempty_buffer_flush_errors;
 
+    // Whether we are in "map mode", i.e. currently in eXamine
+    bool in_map_mode;
```

Comment:
This needs to be initialised in `game_state::game_state()`. There is no default initialisation for struct members, so leaving it uninitialised is undefined behaviour. I think `map_mode_info` itself can be left uninitialised, since you always write to it before reading from it, but it wouldn't hurt to initialise it anyway as a safeguard.